R/text_transform.R

```diff
@@ -146,3 +153,35 @@ text_transform_at_location.cells_column_labels <- function(loc,
 
   data
 }
+
+text_transform_at_location.cells_row_groups <- function(
+    loc,
+    data,
+    fn = identity
+) {
+
+  row_group_vec <- dt_row_groups_get(data = data)
+
+  loc <- to_output_location(loc = loc, data = data)
+
+  for (group in loc$groups) {
+
+    stub_df <- dt_stub_df_get(data = data)
+
+    if (group %in% row_group_vec) {
+
+      if (is.na(group)) next
+
+      stub_df <-
+        stub_df %>%
+        dplyr::mutate(group_label = dplyr::case_when(
```

Comment:
Let's see if we can avoid case_when so that `fn` only needs to be invoked with actual desired groups

Reply:
Now fixed with this:

```r
stub_df[!is.na(stub_df$group_id) & stub_df$group_id == group, ][["group_label"]] <-
  as.list(fn(stub_df[!is.na(stub_df$group_id) & stub_df$group_id == group, ][["group_label"]]))
```